support-frontend/assets/pages/weekly-subscription-checkout/helpers/deliveryDays.ts

```diff
@@ -59,14 +64,54 @@ function getTierThreeDeliveryDate(today?: number) {
 	return result;
 }
 
-function getGuardianAdLiteDate(today?: number) {
-	const firstValidDate = addDays(new Date(today ?? new Date()), 15);
-	return firstValidDate;
-}
+const productDeliveryDate = (
+	productKey: ActiveProductKey,
+	paperProductOptions?: ActivePaperProductOptions,
+): Date | undefined => {
+	console.log(paperProductOptions);
+	switch (productKey) {
+		case 'GuardianAdLite':
+			return addDays(new Date(), 15);
+		case 'TierThree':
+			return getTierThreeDeliveryDate();
+		case 'NationalDelivery':
+		case 'HomeDelivery':
+		case 'SubscriptionCard': {
+			if (paperProductOptions === undefined) {
+				throw new Error(`ratePlan not found for ${productKey}`);
+			}
+			const paperDeliveryDate =
+				productKey === 'SubscriptionCard'
+					? getPaymentStartDate(Date.now(), paperProductOptions)
+					: getHomeDeliveryDays(Date.now(), paperProductOptions)[0];
+			if (paperDeliveryDate === undefined) {
+				throw new Error('delivery date not found for Home Delivery');
```

Comment:
Is `Home Delivery` in this error accurate (looks like this code path could be reached for national delivery or subs card too?

Reply:
What happens if this error gets thrown? Is it caught anywhere? Do we expect this to ever happen?

Reply:
> Is `Home Delivery` in this error accurate (looks like this code path could be reached for national delivery or subs card too?

I think you theoretically can only get an undefined from the HomeDelivery array, hence error message containing `Home Delivery`.

Reply:
> What happens if this error gets thrown? Is it caught anywhere? Do we expect this to ever happen?

Looking at how to use throw, I don't think its necessary to throw these errors at all. Have opted for returning `undefined` for these theoretical possibilities (with an associated comment). Does that sound like a better approach? Updated

Reply:
> Is `Home Delivery` in this error accurate (looks like this code path could be reached for national delivery or subs card too?

That is a superb spot, just checked and should be using `getVoucherDays` for `NationalDelivery`. Thankyou, updated

Reply:
> Does that sound like a better approach? Updated

Great, yes I think that sounds good 👍